tcl/image_file.tcl

```diff
@@ -219,41 +270,81 @@ proc qc::image_cache_create {args} {
     return [dict create width $width height $height file $cache_file]
 }
 
-proc qc::image_data {cache_dir file_id max_width max_height} {
-    #| Return dict of width, height & url of image constrained to max_width & max_height.
+proc qc::image_data {args} {
+    #| Return dict of width, height & url of image,
+    #| constrained to max_width & max_height, optionally auto-cropped
     #| Generates image cache if it doesn't already exist.
-    if { ! [qc::image_cache_exists $cache_dir $file_id $max_width $max_height] } {
-        qc::image_cache_create $cache_dir $file_id $max_width $max_height
+    qc::args $args -autocrop -- cache_dir file_id max_width max_height
+    default autocrop false
+
+    set cache_args [list]
+    if { $autocrop } {
+        lappend cache_args -autocrop
     }
-    return [qc::image_cache_data $cache_dir $file_id $max_width $max_height]
+    lappend cache_args \
+        $cache_dir \
+        $file_id \
+        $max_width \
+        $max_height
+        
+    if { ! [qc::image_cache_exists {*}$cache_args] } {
+        qc::image_cache_create {*}$cache_args
+    }
+    set cache_data [qc::image_cache_data {*}$cache_args]
+
+    return [dict_subset $cache_data url width height]
 }
 
-proc qc::image_handler {cache_dir {error_handler "qc::error_handler"} {image_redirect_handler "qc::image_redirect_handler"} {allowed_max_width 2560} {allowed_max_height 2560} {autocrop false}} {
+proc qc::image_handler {
+    cache_dir
+    {error_handler "qc::error_handler"}
+    {image_redirect_handler "qc::image_redirect_handler"}
+    {allowed_max_width 2560}
+    {allowed_max_height 2560}
+} {
     #| URL handler to serve images that can not be served by fastpath.
     # Create image cache for canonical URL if it doesn't already exist.
-    # If canonical URL was requested return file to client and register URL to be servered by fastpath for future requests.
-    # Otherwise default redirect handler will redirect client to correct image dimesions or the canonical URL.
+    # If canonical URL was requested return file to client,
+    # and register URL to be servered by fastpath for future requests.
```

Comment:
Looks like there were some typos in master. Could you fix them while we're here?

`servered` -> `served`